### PR TITLE
[libffi] Fix compile option -fPIC

### LINKED ISSUE #29660: [at-spi2-atk] Build error
Package: at-spi2-atk[core]:x64-linux -> 2.38.0

**Host Environment**

- Host: x64-linux
- Compiler: GNU 8.3.1
-    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: 25f613278 2023-02-13 (19 hours ago)

**To Reproduce**

`vcpkg install opencv4:x64-linux`

**Failure logs**

```
-- at-spi2-atk currently requires the following libraries from the system package manager:
    libdbus-1

These can be installed on Ubuntu systems via apt-get install libdbus-1-dev
-- Using cached GNOME-at-spi2-atk-AT_SPI2_ATK_2_38_0.tar.gz.
-- Cleaning sources at /home/wpwl/Softwares/vcpkg/buildtrees/at-spi2-atk/src/ATK_2_38_0-a131295237.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/wpwl/Softwares/vcpkg/downloads/GNOME-at-spi2-atk-AT_SPI2_ATK_2_38_0.tar.gz
-- Using source at /home/wpwl/Softwares/vcpkg/buildtrees/at-spi2-atk/src/ATK_2_38_0-a131295237.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
-- Configuring x64-linux-dbg
-- Configuring x64-linux-dbg done
-- Configuring x64-linux-rel
-- Configuring x64-linux-rel done
-- Package x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /usr/bin/ninja install -v
    Working Directory: /home/wpwl/Softwares/vcpkg/buildtrees/at-spi2-atk/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /home/wpwl/Softwares/vcpkg/buildtrees/at-spi2-atk/package-x64-linux-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/at-spi2-atk/portfile.cmake:19 (vcpkg_install_meson)
  scripts/ports.cmake:147 (include)



```
<details><summary>/home/wpwl/Softwares/vcpkg/buildtrees/at-spi2-atk/package-x64-linux-dbg-out.log</summary>

```
[1/37] /opt/rh/devtoolset-8/root/usr/bin/cc -Idroute/libdroute.a.p -Idroute -I../src/ATK_2_38_0-a131295237.clean/droute -I. -I../src/ATK_2_38_0-a131295237.clean -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/glib-2.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -D_DEFAULT_SOURCE -DATK_DISABLE_DEPRECATION_WARNINGS -fPIC -g -fPIC -pthread -MD -MQ droute/libdroute.a.p/droute-pairhash.c.o -MF droute/libdroute.a.p/droute-pairhash.c.o.d -o droute/libdroute.a.p/droute-pairhash.c.o -c ../src/ATK_2_38_0-a131295237.clean/droute/droute-pairhash.c[2/37] /opt/rh/devtoolset-8/root/usr/bin/cc -Idroute/libdroute.a.p -Idroute -I../src/ATK_2_38_0-a131295237.clean/droute -I. -I../src/ATK_2_38_0-a131295237.clean -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/glib-2.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -D_DEFAULT_SOURCE -DATK_DISABLE_DEPRECATION_WARNINGS -fPIC -g -fPIC -pthread -MD -MQ droute/libdroute.a.p/droute-variant.c.o -MF droute/libdroute.a.p/droute-variant.c.o.d -o droute/libdroute.a.p/droute-variant.c.o -c ../src/ATK_2_38_0-a131295237.clean/droute/droute-variant.c
[3/37] /opt/rh/devtoolset-8/root/usr/bin/cc -Iatk-adaptor/adaptors/libatk-bridge-adaptors.a.p -Iatk-adaptor/adaptors -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor/adaptors -I. -I../src/ATK_2_38_0-a131295237.clean -Iatk-adaptor -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/at-spi-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/glib-2.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/atk-1.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -D_DEFAULT_SOURCE -DATK_DISABLE_DEPRECATION_WARNINGS -fPIC -g -fPIC -pthread -MD -MQ atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/document-adaptor.c.o -MF atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/document-adaptor.c.o.d -o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/document-adaptor.c.o -c ../src/ATK_2_38_0-a131295237.clean/atk-adaptor/adaptors/document-adaptor.c
...
Skipped 87 lines
...

[28/37] /opt/rh/devtoolset-8/root/usr/bin/cc -Iatk-adaptor/libatk-bridge-2.0.a.p -Iatk-adaptor -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor -I. -I../src/ATK_2_38_0-a131295237.clean -Iatk-adaptor/adaptors -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor/adaptors -Idroute -I../src/ATK_2_38_0-a131295237.clean/droute -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/at-spi-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/glib-2.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/atk-1.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -D_DEFAULT_SOURCE -DATK_DISABLE_DEPRECATION_WARNINGS -fPIC -g -fPIC -pthread -MD -MQ atk-adaptor/libatk-bridge-2.0.a.p/accessible-register.c.o -MF atk-adaptor/libatk-bridge-2.0.a.p/accessible-register.c.o.d -o atk-adaptor/libatk-bridge-2.0.a.p/accessible-register.c.o -c ../src/ATK_2_38_0-a131295237.clean/atk-adaptor/accessible-register.c
[29/37] /opt/rh/devtoolset-8/root/usr/bin/cc  -o droute/droute-test droute/droute-test.p/droute-test.c.o -L/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group droute/libdroute.a -pthread /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a -lm /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libpcre2-8.a /usr/lib64/libdbus-1.so /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libatspi.a /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libgobject-2.0.a /usr/lib64/libX11.so /usr/lib64/libXtst.so /usr/lib64/libXi.so /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libffi.a -Wl,--end-group
[30/37] /opt/rh/devtoolset-8/root/usr/bin/cc -Iatk-adaptor/libatk-bridge-2.0.a.p -Iatk-adaptor -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor -I. -I../src/ATK_2_38_0-a131295237.clean -Iatk-adaptor/adaptors -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor/adaptors -Idroute -I../src/ATK_2_38_0-a131295237.clean/droute -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/at-spi-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/glib-2.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/atk-1.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -D_DEFAULT_SOURCE -DATK_DISABLE_DEPRECATION_WARNINGS -fPIC -g -fPIC -pthread -MD -MQ atk-adaptor/libatk-bridge-2.0.a.p/accessible-stateset.c.o -MF atk-adaptor/libatk-bridge-2.0.a.p/accessible-stateset.c.o.d -o atk-adaptor/libatk-bridge-2.0.a.p/accessible-stateset.c.o -c ../src/ATK_2_38_0-a131295237.clean/atk-adaptor/accessible-stateset.c
[31/37] /opt/rh/devtoolset-8/root/usr/bin/cc -Iatk-adaptor/libatk-bridge-2.0.a.p -Iatk-adaptor -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor -I. -I../src/ATK_2_38_0-a131295237.clean -Iatk-adaptor/adaptors -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor/adaptors -Idroute -I../src/ATK_2_38_0-a131295237.clean/droute -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/at-spi-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/glib-2.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/atk-1.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -D_DEFAULT_SOURCE -DATK_DISABLE_DEPRECATION_WARNINGS -fPIC -g -fPIC -pthread -MD -MQ atk-adaptor/libatk-bridge-2.0.a.p/object.c.o -MF atk-adaptor/libatk-bridge-2.0.a.p/object.c.o.d -o atk-adaptor/libatk-bridge-2.0.a.p/object.c.o -c ../src/ATK_2_38_0-a131295237.clean/atk-adaptor/object.c
[32/37] /opt/rh/devtoolset-8/root/usr/bin/cc -Iatk-adaptor/libatk-bridge-2.0.a.p -Iatk-adaptor -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor -I. -I../src/ATK_2_38_0-a131295237.clean -Iatk-adaptor/adaptors -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor/adaptors -Idroute -I../src/ATK_2_38_0-a131295237.clean/droute -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/at-spi-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/glib-2.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/atk-1.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -D_DEFAULT_SOURCE -DATK_DISABLE_DEPRECATION_WARNINGS -fPIC -g -fPIC -pthread -MD -MQ atk-adaptor/libatk-bridge-2.0.a.p/spi-dbus.c.o -MF atk-adaptor/libatk-bridge-2.0.a.p/spi-dbus.c.o.d -o atk-adaptor/libatk-bridge-2.0.a.p/spi-dbus.c.o -c ../src/ATK_2_38_0-a131295237.clean/atk-adaptor/spi-dbus.c
[33/37] /opt/rh/devtoolset-8/root/usr/bin/cc -Iatk-adaptor/libatk-bridge-2.0.a.p -Iatk-adaptor -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor -I. -I../src/ATK_2_38_0-a131295237.clean -Iatk-adaptor/adaptors -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor/adaptors -Idroute -I../src/ATK_2_38_0-a131295237.clean/droute -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/at-spi-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/glib-2.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/atk-1.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -D_DEFAULT_SOURCE -DATK_DISABLE_DEPRECATION_WARNINGS -fPIC -g -fPIC -pthread -MD -MQ atk-adaptor/libatk-bridge-2.0.a.p/bridge.c.o -MF atk-adaptor/libatk-bridge-2.0.a.p/bridge.c.o.d -o atk-adaptor/libatk-bridge-2.0.a.p/bridge.c.o -c ../src/ATK_2_38_0-a131295237.clean/atk-adaptor/bridge.c
[34/37] /opt/rh/devtoolset-8/root/usr/bin/cc -Iatk-adaptor/libatk-bridge-2.0.a.p -Iatk-adaptor -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor -I. -I../src/ATK_2_38_0-a131295237.clean -Iatk-adaptor/adaptors -I../src/ATK_2_38_0-a131295237.clean/atk-adaptor/adaptors -Idroute -I../src/ATK_2_38_0-a131295237.clean/droute -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/at-spi-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/glib-2.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/glib-2.0/include -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../../include/atk-1.0 -I/home/wpwl/Softwares/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -O0 -g -D_DEFAULT_SOURCE -DATK_DISABLE_DEPRECATION_WARNINGS -fPIC -g -fPIC -pthread -MD -MQ atk-adaptor/libatk-bridge-2.0.a.p/event.c.o -MF atk-adaptor/libatk-bridge-2.0.a.p/event.c.o.d -o atk-adaptor/libatk-bridge-2.0.a.p/event.c.o -c ../src/ATK_2_38_0-a131295237.clean/atk-adaptor/event.c
[35/37] rm -f atk-adaptor/libatk-bridge-2.0.a && /opt/rh/devtoolset-8/root/usr/bin/ar csrD atk-adaptor/libatk-bridge-2.0.a atk-adaptor/libatk-bridge-2.0.a.p/accessible-leasing.c.o atk-adaptor/libatk-bridge-2.0.a.p/accessible-cache.c.o atk-adaptor/libatk-bridge-2.0.a.p/accessible-register.c.o atk-adaptor/libatk-bridge-2.0.a.p/accessible-stateset.c.o atk-adaptor/libatk-bridge-2.0.a.p/introspection.c.o atk-adaptor/libatk-bridge-2.0.a.p/bridge.c.o atk-adaptor/libatk-bridge-2.0.a.p/object.c.o atk-adaptor/libatk-bridge-2.0.a.p/event.c.o atk-adaptor/libatk-bridge-2.0.a.p/spi-dbus.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/accessible-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/action-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/application-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/cache-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/collection-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/component-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/document-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/editabletext-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/hyperlink-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/hypertext-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/image-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/selection-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/socket-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/table-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/table-cell-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/text-adaptor.c.o atk-adaptor/adaptors/libatk-bridge-adaptors.a.p/value-adaptor.c.o droute/libdroute.a.p/droute.c.o droute/libdroute.a.p/droute-variant.c.o droute/libdroute.a.p/droute-pairhash.c.o
[36/37] /opt/rh/devtoolset-8/root/usr/bin/cc  -o atk-adaptor/libatk-bridge.so atk-adaptor/libatk-bridge.so.p/gtk-2.0_module.c.o -L/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib -Wl,--as-needed -Wl,--allow-shlib-undefined -shared -fPIC -Wl,--start-group atk-adaptor/libatk-bridge-2.0.a atk-adaptor/adaptors/libatk-bridge-adaptors.a droute/libdroute.a /usr/lib64/libdbus-1.so -pthread /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libatk-1.0.a /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libgobject-2.0.a /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a -lm /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libffi.a /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libpcre2-8.a /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libatspi.a /usr/lib64/libX11.so /usr/lib64/libXtst.so /usr/lib64/libXi.so -Wl,--export-dynamic /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libgmodule-2.0.a -ldl -Wl,--end-group
FAILED: atk-adaptor/libatk-bridge.so 
/opt/rh/devtoolset-8/root/usr/bin/cc  -o atk-adaptor/libatk-bridge.so atk-adaptor/libatk-bridge.so.p/gtk-2.0_module.c.o -L/home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib -Wl,--as-needed -Wl,--allow-shlib-undefined -shared -fPIC -Wl,--start-group atk-adaptor/libatk-bridge-2.0.a atk-adaptor/adaptors/libatk-bridge-adaptors.a droute/libdroute.a /usr/lib64/libdbus-1.so -pthread /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libatk-1.0.a /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libgobject-2.0.a /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libglib-2.0.a -lm /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libffi.a /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libpcre2-8.a /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libatspi.a /usr/lib64/libX11.so /usr/lib64/libXtst.so /usr/lib64/libXi.so -Wl,--export-dynamic /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libgmodule-2.0.a -ldl -Wl,--end-group
/opt/rh/devtoolset-8/root/usr/libexec/gcc/x86_64-redhat-linux/8/ld: /home/wpwl/Softwares/vcpkg/installed/x64-linux/debug/lib/pkgconfig/../../lib/libffi.a(unix64.S.o): relocation R_X86_64_PC32 against symbol `abort@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/opt/rh/devtoolset-8/root/usr/libexec/gcc/x86_64-redhat-linux/8/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix https://github.com/microsoft/vcpkg/issues/29660

cc @HoMiGT @872448677 @dominique120  Please test this. 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libffi/CMakeLists.txt [hunk 1]
```diff
@@ -214,6 +214,10 @@ if(CMAKE_STATIC_LIBRARY_PREFIX STREQUAL "lib")
     set_target_properties(libffi PROPERTIES OUTPUT_NAME ffi)
 endif()
 
+if(NOT WIN32)
+    target_compile_options(libffi PUBLIC -fPIC)
+endif()
+
```
Comment (Neumann-A) [inline ports/libffi/CMakeLists.txt:220]: Hmm no that should come from the toolchain.

Comment (FrankXie05) [reply ports/libffi/CMakeLists.txt]: Done.

**Comment by Neumann-A** [comment]
Not really: 
https://github.com/microsoft/vcpkg/blob/576f3f1ad13bf2c3bd06520adfe41b69448cc033/scripts/toolchains/linux.cmake#L52-L53

So the -fpic flags should already be there

**Comment by Neumann-A** [comment]
and according to https://github.com/microsoft/vcpkg/issues/29660#issuecomment-1430996709 the problem ist `CMAKE_ASM_FLAGS` not having `-fPIC`

File: ports/libffi/CMakeLists.txt [hunk 2]
```diff
@@ -214,6 +214,10 @@ if(CMAKE_STATIC_LIBRARY_PREFIX STREQUAL "lib")
     set_target_properties(libffi PROPERTIES OUTPUT_NAME ffi)
 endif()
 
+if(NOT WIN32)
+    string(APPEND CMAKE_ASM_FLAGS " -fPIC")
```
Comment (FrankXie05) [inline ports/libffi/CMakeLists.txt:218]: Needs to test.

**Comment by dg0yt** [comment]
> and according to [#29660 (comment)](https://github.com/microsoft/vcpkg/issues/29660#issuecomment-1430996709) the problem ist `CMAKE_ASM_FLAGS` not having `-fPIC`

My conclusion in that comment wrong. What I learned from looking into this issue is that `-fPIC` should only affect what kind of asm code is generated from C or C++, but it shouldn't have any affect on what is already asm.

**Comment by FrankXie05** [comment]
Even if `project(libffi C ASM)` is used, it doesn't mean that all code in the project can automatically adapt to the requirement of generating position independent code. If this doesn't help, maybe we need to dig into the file `unix64.S`.🫤

**Comment by dg0yt** [comment]
> the requirement of generating position independent code.

AFAIU it is not about *generating* (with `-fPIC` from C or C++) but about *having* position independent asm code (and *having this recognized* by the gcc linker).

Maybe it depends on proper detection of `HAVE_AS_X86_PCREL`?

**Comment by FrankXie05** [comment]
`PC-relative` addressing for `x86`? Maybe that is the key.

**Comment by dominique120** [comment]
@FrankXie05 

libffi builds fine, however when I try to build at-api2-atk it still fails:
image

Here is the log for the build:
[package-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/12344183/package-x64-linux-dbg-out.log)

libffi tested on AMI Linux 2 - working:
image

**Comment by FrankXie05** [comment]
@dominique120 Could you please add the follow code to your `ports/libffi/CMakeLists.txt` and try again?
```
set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

**Comment by dg0yt** [comment]
> @dominique120 Could you please add the follow code to your `ports/libffi/CMakeLists.txt` and try again?
> 
> ```
> set(CMAKE_POSITION_INDEPENDENT_CODE ON)
> ```

IIRC I already tested that, and it doesn't help. And a third time, it is not expected to have an effect on asm code.

If the vendored build system can't be maintained due to lack of know-how, it shouldn't be used. https://github.com/microsoft/vcpkg/pull/33203

**Comment by FrankXie05** [comment]
I am in favor of using the official build system. But for this issue, the upstream only has macros for `x86`.
https://github.com/winlibs/libffi/blob/921491ac934bc3ce4ef3fc80b553c609d0f1fcd9/fficonfig.h#L59

**Comment by FrankXie05** [comment]
Closed this PR.

**Pull request closed without merging**